plugins/source/mixpanel/client/client.go

```diff
@@ -0,0 +1,116 @@
+package client
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net/http"
+	"time"
+
+	"github.com/cloudquery/cloudquery/plugins/source/mixpanel/internal/mixpanel"
+	"github.com/cloudquery/plugin-sdk/backend"
+	"github.com/cloudquery/plugin-sdk/plugins/source"
+	"github.com/cloudquery/plugin-sdk/schema"
+	"github.com/cloudquery/plugin-sdk/specs"
+	"github.com/rs/zerolog"
+)
+
+type Client struct {
+	logger     zerolog.Logger
+	sourceSpec specs.Source
+
+	MPSpec   Spec
+	Services *mixpanel.Client
+	Backend  backend.Backend
+}
+
+func New(logger zerolog.Logger, sourceSpec specs.Source, mpSpec Spec, services *mixpanel.Client, bk backend.Backend) Client {
+	return Client{
+		logger:     logger,
+		sourceSpec: sourceSpec,
+		MPSpec:     mpSpec,
+		Services:   services,
+		Backend:    bk,
+	}
+}
+
+func (c *Client) Logger() *zerolog.Logger {
+	return &c.logger
+}
+
+func (c *Client) ID() string {
+	return c.sourceSpec.Name
+}
+
+func Configure(ctx context.Context, logger zerolog.Logger, s specs.Source, o source.Options) (schema.ClientMeta, error) {
+	mpSpec := &Spec{}
+	if err := s.UnmarshalSpec(mpSpec); err != nil {
+		return nil, fmt.Errorf("failed to unmarshal mixpanel spec: %w", err)
+	}
+
+	services, err := getServiceClient(logger, mpSpec)
+	if err != nil {
+		return nil, err
+	}
+
+	cl := New(logger, s, *mpSpec, services, o.Backend)
+	return &cl, nil
+}
+
+func getServiceClient(logger zerolog.Logger, spec *Spec) (*mixpanel.Client, error) {
+	if spec.Secret == "" {
+		// allow empty username for the deprecated project secret method: https://developer.mixpanel.com/reference/project-secret
+		return nil, errors.New("no credentials provided")
+	}
+	if spec.ProjectID < 1 {
+		return nil, errors.New("no project id provided")
+	}
+
+	const dateFormat = "2006-01-02"
+
+	if spec.StartDate == "" {
+		dt := time.Now().UTC().Add(-30 * 86400 * time.Second).Format(dateFormat)
+		logger.Info().Str("start_date", dt).Msg("no start date provided, defaulting to 30 days ago")
+		spec.StartDate = dt
+	}
+	if spec.EndDate == "" {
+		dt := time.Now().UTC().Add(-86400 * time.Second).Format(dateFormat)
+		logger.Info().Str("end_date", dt).Msg("no end date provided, defaulting to yesterday")
```

Comment:
Why yesterday and not today?

Reply:
Usually (historically, based on previous multiple analytics and marketing data fetching experience) today's data is either incomplete (bad if you're using cursors and don't account for it) or doesn't exist... I didn't test this with mixpanel but it seemed like a good compromise. A way to specify relative dates could work here (maybe if it's an integer we can treat them as days to go back, etc...)

Reply:
I think syncing today's data should be fine, as on the next sync you will fill in the late-arriving datapoints. Incremental syncs should take care of this by fetching some overlapping window of events again on the next sync (probably about an hour), but it shouldn't be a concern for non-incremental tables.

Reply:
Looks like the incremental cursor is going to be the `time` field (which is unix seconds). So we could move this to be today, if other resources don't fail if today is requested. If they do, we'll just fallback to the sane default of yesterday for those resources (can have a separate EndDateExceptToday field that gets auto calculated and so on)

Reply:
None of the resources complained, so now the end date is today.